### PR TITLE
Module 11

### PR DESCRIPTION
**Pull request by SamSeed-Softwire**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,30 +27,28 @@ jobs:
...
     deploy:
       provider: script
       script: bash ./deploy_prod.sh
       skip_cleanup: true
     after_deploy:
-    - curl "https://$HEROKU_APP_NAME.herokuapp.com/"
-    - curl_response=$(curl "https://$HEROKU_APP_NAME.herokuapp.com/")
-    - if [[ "$curl_response" == *"To-Do App"* ]]; then echo "Successful curl!"; else >&2 echo "Curl failure :("; return 1; fi
+    - curl_response=$(curl "http://$AZURE_APP_SERVICE_NAME.azurewebsites.net/")
+    # TODO: cause Travis build to fail if the curl is unsuccessful.
```
Comment (hugh-emerson) [inline .travis.yml]: This is a good idea but Travis ignores the exit code of after_deploy. So the correct solution may be a separate "stage".

Also, I'm not sure but you may need a short delay to really be sure you're hitting the latest version and it's had time to deploy.

**Review by hugh-emerson — CHANGES_REQUESTED** [review]
Seems reasonable but there are two issues I'm unsure about:

The `SoftwirePilot_SamSeed_ProjectExercise` resource group only contains a DB. Where's your app service? I don't see any called `softwirepilot-samseed-projectexercise-app-service` which is what your Travis build seems to expect. I assume you did have it running at some point but if it's been deleted, can you recreate it?

Secondly, your commit says you've changed Travis to only build on pull requests. You don't want that - deployments will only happen for pushes to master. They don't (and shouldn't) run for pull requests.

**Comment by SamSeed-Softwire** [comment]
Thanks @hugh-emerson. I think my app service got deleted when my free tier expired... I've recreated it using a new B1 tier app service plan (it wouldn't let me recreate a free tier app service plan, unsurprisingly!)

File: .travis.yml (hunk 1 above)
Comment (SamSeed-Softwire) [reply .travis.yml]: Moving to a separate stage and putting the curl response check in a 'script' phase made this work, thanks for the idea! Also, Travis takes long enough to start a new stage that I essentially get a decent delay after the deployment for free.

File: .travis.yml [hunk 2]
```diff
@@ -27,30 +25,31 @@ jobs:
...
-    after_success:
-    - docker tag samseedsoftwire/todo-app-prod:latest registry.heroku.com/$HEROKU_APP_NAME/web
-    before_deploy:
-    - heroku container:login --verbose
+    - skip
     deploy:
       provider: script
       script: bash ./deploy_prod.sh
+      on:
+        branch: module_11
```
Comment (hugh-emerson) [inline .travis.yml]: Don't forget to revert this

**Review by hugh-emerson — APPROVED** [review]
Looks good! 👍

**Pull request merged**